app/scripts/migrations/092.3.ts

```diff
@@ -0,0 +1,92 @@
+import { isNullOrUndefined, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+import log from 'loglevel';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 92.3;
+
+/**
+ * This migration does the following:
+ *
+ * - Deletes currently stored advancedGasFee in preferences controller,
+ * replacing the default with an empty object
+ * - Sets hadAdvancedGasFeesSetPriorToMigration92_3 flag on AppStateController
+ * to indicate if the user had previously had advancedGasFee set in their
+ * preferences. This will be used to display a whats new entry to inform users
+ * that we wiped these settings and made them apply per network.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  migrateData(versionedData.data);
+  return versionedData;
+}
+
+function migrateData(state: Record<string, unknown>): void {
+  changeShapeAndRemoveOldAdvancedGasFeePreference(state);
+}
+
+function changeShapeAndRemoveOldAdvancedGasFeePreference(
+  state: Record<string, any>,
```

Comment:
The use of `any` here is disabling type checking. Could we use `unknown` here instead?

Reply:
Absolutely